Subject: [PATCH] Remove questionable CMake config
---
Index: Sources/ComplexModule/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Sources/ComplexModule/CMakeLists.txt b/Sources/ComplexModule/CMakeLists.txt
--- a/Sources/ComplexModule/CMakeLists.txt	(revision 8c4747f116cb49ea83a2fec954f9668547fdf1cc)
+++ b/Sources/ComplexModule/CMakeLists.txt	(revision 7675331540c9b0260e4638b73440c326ea9ca8c8)
@@ -19,8 +19,6 @@
   Complex+StringConvertible.swift
   Polar.swift
   Scale.swift)
-set_target_properties(ComplexModule PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
 target_link_libraries(ComplexModule PUBLIC
   RealModule)
 
Index: Sources/Numerics/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Sources/Numerics/CMakeLists.txt b/Sources/Numerics/CMakeLists.txt
--- a/Sources/Numerics/CMakeLists.txt	(revision 8c4747f116cb49ea83a2fec954f9668547fdf1cc)
+++ b/Sources/Numerics/CMakeLists.txt	(revision 7675331540c9b0260e4638b73440c326ea9ca8c8)
@@ -9,8 +9,7 @@
 
 add_library(Numerics
   Numerics.swift)
-set_target_properties(Numerics PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
+
 # NOTE: generate the force load symbol to ensure that the import library is
 # generated on Windows for autolinking.
 target_compile_options(Numerics PUBLIC
Index: Sources/RealModule/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Sources/RealModule/CMakeLists.txt b/Sources/RealModule/CMakeLists.txt
--- a/Sources/RealModule/CMakeLists.txt	(revision 8c4747f116cb49ea83a2fec954f9668547fdf1cc)
+++ b/Sources/RealModule/CMakeLists.txt	(revision 7675331540c9b0260e4638b73440c326ea9ca8c8)
@@ -19,8 +19,6 @@
   Real.swift
   RealFunctions.swift
   RelaxedArithmetic.swift)
-set_target_properties(RealModule PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
 target_link_libraries(RealModule PUBLIC
   _NumericsShims)
 
Index: Sources/_TestSupport/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Sources/_TestSupport/CMakeLists.txt b/Sources/_TestSupport/CMakeLists.txt
--- a/Sources/_TestSupport/CMakeLists.txt	(revision 8c4747f116cb49ea83a2fec954f9668547fdf1cc)
+++ b/Sources/_TestSupport/CMakeLists.txt	(revision 7675331540c9b0260e4638b73440c326ea9ca8c8)
@@ -13,8 +13,6 @@
   Error.swift
   Interval.swift
   RealTestSupport.swift)
-set_target_properties(_TestSupport PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
 target_link_libraries(_TestSupport PUBLIC
   Numerics)
 
Index: Tests/ComplexTests/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Tests/ComplexTests/CMakeLists.txt b/Tests/ComplexTests/CMakeLists.txt
--- a/Tests/ComplexTests/CMakeLists.txt	(revision 8c4747f116cb49ea83a2fec954f9668547fdf1cc)
+++ b/Tests/ComplexTests/CMakeLists.txt	(revision 7675331540c9b0260e4638b73440c326ea9ca8c8)
@@ -12,8 +12,7 @@
   ArithmeticTests.swift
   ElementaryFunctionTests.swift
   PropertyTests.swift)
-set_target_properties(ComplexTests PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
+
 target_compile_options(ComplexTests PRIVATE
   -enable-testing)
 target_link_libraries(ComplexTests PUBLIC
